src/JobPanel/backend/_service_proxy.py

```diff
@@ -70,22 +73,34 @@ def start_service(self):
         - send port RYY, wait for OK
         - after OK, close starting connection
         """
-        pass
+
+        # 4.
+        delegator = self.connection.get_delegator()
 
 
-    def connect_service(self, service_port=None):
+    def connect_service(self, child_id):
         """
         Assumes Job is running and listening on given remote port. (we get port either throuhg initial socket
         connection or by reading the remote config file - reinit part of __init__)
 
-        If service_port is None, try to download service_data from remote and get the listening port from it.
-
-        - local forwarding tunnel,  port LYY to port RYY
-        - create final connection, self.repeater
-        - set Job state to running
-        (repeater starts appropriate dispatcher, automatically reporting problem with connecting after some timeout using special answer).
+        1. Check if repeater.client[child_id] have target port (ClientDispatcher exists)
+        2. If No or child_id == None, download service config file and get port from there ( ... postpone)
+        3. Open forward tunnel
+        4. Call repeater.client[id].connect(port)
+           self.repeater._connect_child_repeater(socket_address)
+        5. set Job state to running
         """
-        pass
+
+        # 1.
+        if self.repeater.clients[child_id] is None:
+            return
+
+        # 3.
+        local_port = self.connection.forward_local_port(self.repeater.clients[child_id].address[1])
+        print(local_port)
+
+        # 4.
+        self.repeater.clients[child_id].my_connect(("localhost", local_port))
```

Comment:
In order to keep access to async_repeater thread safe, we should rather not access its members directly, but only through properly designed access methods. 
AsyncRepeater._connect_child_repeater was meant to this end.